synapseclient/extensions/curator/schema_generation.py

```diff
@@ -4329,34 +4412,14 @@ def _get_validation_rule_based_fields(
 
         js_is_array = ValidationRuleName.LIST in validation_rule_names
 
-        # The explicit JSON Schema type is the one set in the data model
         # The implicit JSON Schema type is the one implied by the presence
         #   of certain validation rules
-        # Schematic will use the implicit type if the explicit type isn't specified for now,
-        #   but this behavior is deprecated and will be removed in the future by SCHEMATIC-326
+
         implicit_js_type = get_js_type_from_inputted_rules(validation_rules)
-        # If there is an explicit and implicit type set and the implicit type conflicts with the
-        # explicit type, then an exception is raised
-        if (
-            explicit_js_type
-            and implicit_js_type
-            and explicit_js_type != implicit_js_type
-        ):
-            msg = (
-                f"Property: '{name}', has explicit type: '{explicit_js_type}' "
-                f"that conflicts with the implicit type: '{implicit_js_type}' "
-                f"derived from its validation rules: {validation_rules}"
-            )
-            logger.warning(msg)
-        if not explicit_js_type and implicit_js_type:
-            js_type = implicit_js_type
-            msg = (
-                f"No explicit type set for property: '{name}', "
-                "using validation rules to set the type. "
-                "Using validation rules to set type is deprecated. "
-                "You should set the columnType for this property in your data model."
+        if implicit_js_type:
+            logger.warning(
+                f"Detected implicit type: '{implicit_js_type}' for property: '{name}'. Please note that type can only be set explicitly via the columnType column in the data model going forward."
```

Comment:
We've already deprecated using type validaiton rules, you get rid of anything to do with checking type here.